Sources/SwiftBuildSupport/PackagePIFProjectBuilder+Modules.swift

```diff
@@ -488,14 +488,16 @@ extension PackagePIFProjectBuilder {
         if enableDuplicateLinkageCulling {
             impartedSettings[.LD_WARN_DUPLICATE_LIBRARIES] = "NO"
         }
-        impartedSettings[.OTHER_LDFLAGS] = (sourceModule.isCxx ? ["-lc++"] : []) + ["$(inherited)"]
-        impartedSettings[.OTHER_LDRFLAGS] = []
-        log(
-            .debug,
-            indent: 1,
-            "Added '\(impartedSettings[.OTHER_LDFLAGS]!)' to imparted OTHER_LDFLAGS"
-        )
-
+        if sourceModule.isCxx {
+            for platform in ProjectModel.BuildSettings.Platform.allCases {
+                // darwin & freebsd
+                if [.macOS, .macCatalyst, .iOS, .watchOS, .tvOS, .xrOS, .driverKit, .freebsd].contains(platform) {
+                    settings[.OTHER_LDFLAGS, platform] = ["-lc++", "$(inherited)"]
+                } else if platform != .windows { // all others except windows
```

Comment:
I wonder if we should warn or emit a debug log if we encounter a totally unknown platform here so we know we need to go fix it up